### PR TITLE
Dockerize the app

### PR DESCRIPTION
**Pull request by jmhansen**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Pull base image
+FROM python:3.6-slim
+
+# Set environment varibles
+ENV PYTHONUNBUFFERED 1
+ENV PYTHONUNBUFFERED 1
```
Comment (longjos) [inline Dockerfile]: Doubly sure it's set? :)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+services:
+  db:
+    image: postgres:alpine
+    environment:
+      POSTGRES_DB: 'smile_widget_db'
+      POSTGRES_USER: 'postgres'
+      POSTGRES_PASSWORD: 'postgres'
+    volumes:
+      - postgres_data:/var/lib/postgresql/data/
```
Comment (baronvonvaderham) [inline docker-compose.yml]: I don't think this is strictly necessary and may have issues on some users' systems to specify an absolute path like this. We probably don't really need data persistence if the users use "docker-compose down".

File: Dockerfile (hunk 1 above)
Comment (jmhansen) [reply Dockerfile]: o...this was supposed to be `ENV PYTHONDONTWRITEBYTECODE 1`, which I stole from this guide: https://wsvincent.com/django-docker-postgresql/.  Apparently it prevents Python from writing `.pyc` files.

File: wait-for-postgres.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+set -e
+
+>&2 echo "Beginning startup script"
+>&2 echo "Postgres user is $POSTGRES_USER"
+>&2 echo "Postgres DB is $POSTGRES_DB"
+
+until PGPASSWORD=$POSTGRES_PASSWORD psql -U "$POSTGRES_USER" "$POSTGRES_DB" -c '\q'; do
+  >&2 echo "Postgres is unavailable - sleeping"
+  sleep 1
```
Comment (baronvonvaderham) [inline wait-for-postgres.sh]: I think this is the loop I'm going to enter this weekend with my brain.

**Pull request merged**